modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountEntryLocalServiceImpl.java

```diff
@@ -451,11 +451,25 @@ public List<AccountEntry> getAccountEntries(
 			companyId, status, start, end, orderByComparator);
 	}
 
+	@Override
+	public List<AccountEntry> getAccountEntries(
+		long companyId, String type, int status, int start, int end,
+		OrderByComparator<AccountEntry> orderByComparator) {
+
+		return accountEntryPersistence.findByC_T_S(
```

Comment:
A/C talk about Approved and not deactivated account, where is the active clause?

Reply:
As far as I know the status of the AccountEntry is managed by the workflow statuses (and the active/approved status is managed by the STATUS_APPROVED status).
Also, in the AccountEntryLocalServiceImpl I can see that the methods activateAccountEntry / deactivateAccountEntry work updating the status attribute
```
@Override
public AccountEntry activateAccountEntry(AccountEntry accountEntry)
	throws PortalException {

	return updateStatus(accountEntry, WorkflowConstants.STATUS_APPROVED);
}
```

Reply:
have you tried to use com.liferay.account.service.impl.AccountEntryLocalServiceImpl#searchAccountEntries ?
that goes through the index and should account also for permission checking.
You just need to set the type in the `params`

Reply:
Replaced with the AccountEntryLocalServiceImpl#searchAccountEntries method so the new finder is not needed anymore